docs/docs/3-features-modules/10-warning-module/index.md

```diff
@@ -0,0 +1,278 @@
+    ) {}
+
+    async createWarnings(block: NewsBlockData): Promise<BlockWarning[]> {
+        const warnings: BlockWarning[] = [];
+
+        if (block.id) {
+            const news = await this.newsRepository.findOne({ id: block.id });
+            if (!news) {
+                warnings.push({ severity: "high", message: "invalidNewsTarget" });
+            }
+        } else {
+            warnings.push({ severity: "high", message: "missingNewsTarget" });
+        }
+        return warnings;
+    }
+}
+```
+
+## API: Custom Entity Warnings
+
+You can also create warnings for entities.  
+To do this, use the `@CreateWarning` decorator
+
+### Simple Entity Warning
+
+```typescript
+@CreateWarnings<News>(async (value) => {
+    if (!value.title) {
+        return [
+            {
+                message: "missingTitle",
+                severity: "low",
+            },
+        ];
+    }
+    return [];
+})
+@Entity()
+class News {
+    @Field({ nullable: true })
+    @Property({ columnType: "text", nullable: true })
+    title?: string;
+}
+```
+
+### Entity Warning with dependency injection
+
+#### Example of the Entity:
+
+```typescript
+@CreateWarning(NewsWarningService)
+@Entity()
+class NewsPageType {
+    @Field({ nullable: true })
+    @Property({ columnType: "text", nullable: true })
+    newsId?: string;
+}
+```
+
+#### Example of the Service:
+
+```typescript
+class NewsWarningService implements CreateWarningsServiceInterface<NewsPageType> {
+    constructor(
+        @InjectRepository(News)
+        private readonly newsRepository: EntityRepository<News>,
+    ) {}
+
+    async createWarnings(entity: MyEntity): Promise<EntityWarning[]> {
+        const warnings: EntityWarning[] = [];
+
+        if (entity.newsId) {
+            const news = await this.newsRepository.findOne(entity.newsId);
+            if (!news) {
+                warnings.push({ severity: "high", message: "invalidNewsTarget" });
+            }
+        }
+
+        return warnings;
+    }
+}
+```
+
+### Entity Warning for improved performance
+
+When defining a warning with one of the above methods, performance issues may arise if your entity's database table contains numerous entries. In such cases, the `WarningService` provides a generator function `async *bulkCreateWarnings()` that allows for more efficient querying and checking of warnings.
+
+Instead of checking all records (which happens with the regular `createWarnings` function), you can define specific criteria for which entities should be checked. For example, if you only need to check warnings for active `News` items (excluding archived ones), you can reduce the number of records to check.
+
+Here's how to implement this:
+
+```typescript
+class NewsWarningService implements CreateWarningsServiceInterface {
+    constructor(
+        @InjectRepository(News)
+        private readonly newsRepository: EntityRepository<News>,
+    ) {}
+
+    async *bulkCreateWarnings() {
```

Comment:
Is this always a generator function? Or is it possible to return a plain array as well?

Reply:
Currently it's only possible to define a generator function.